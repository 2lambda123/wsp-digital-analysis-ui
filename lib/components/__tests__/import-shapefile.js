/* global describe, it, expect, jest */

import renderer from 'react-test-renderer'
import React from 'react'

import ImportShapefile from '../import-shapefile'

describe('Component > ImportShapefile', () => {
  it('renders correctly', () => {
    const props = {
      close: jest.fn(),
      setModification: jest.fn(),
      variants: [],
      scenarioId: '1'
    }

    // mount component
    const tree = renderer.create(
      <ImportShapefile
        {...props}
        />
    ).toJSON()
    expect(tree).toMatchSnapshot()
    const noCalls = [
      'close',
      'setModification'
    ]
    noCalls.forEach((fn) => {
      expect(props[fn]).not.toBeCalled()
    })
  })
})
