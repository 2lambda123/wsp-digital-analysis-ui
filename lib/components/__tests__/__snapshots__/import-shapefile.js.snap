// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component > ImportShapefile renders correctly 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading clearfix "
  >
    Import Shapefile
  </div>
  <div
    className="panel-body"
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="Select Shapefile (zipped):"
      >
        Select Shapefile (zipped):
      </label>
      <input
        className="form-control"
        name={undefined}
        onChange={[Function]}
        placeholder="Select Shapefile (zipped):"
        type="file"
        value={undefined}
      />
    </div>
    <a
      className="btn btn-success btn-block"
      disabled={true}
      href="#"
      onClick={[Function]}
      tabIndex={0}
      target={undefined}
    >
      Import
    </a>
    <a
      className="btn btn-danger btn-block"
      href="#"
      onClick={[Function]}
      tabIndex={0}
      target={undefined}
    >
      Cancel
    </a>
  </div>
</div>
`;
